00-framework-01-maturity.md

```diff
@@ -0,0 +1,118 @@
+  * motivation: why this specification is relevant, and how it advances the
+    status quo.
+  * scope and rationale: what areas of the technical system the specification
+    impacts.
+  * goals: what we expect to achieve (positively and negatively) as a result
+    of implementing the specification.
+  * expected feature set: a summary/enumeration of features the spec provides.
+  * tentative technical directions: how are we planning to materialise the
+    specification in terms of system design.
+
+The `Initial Working Draft` shall be reviewed by the community in no more than 3
+working days. Should there be no defects in form, content or serious technical
+soundness issues, the `Initial Working Draft` will be accepted and merged.
+
+Ideas deemed controversial or breaking, and those that garner subjective
+opposition, will still be accepted in order to give them a venue to grow,
+mature and iterate.
+
+Once the `Initial Working Draft` is merged, the author may continue revising
+and evolving their specification by self-approving their own *Pull Requests*.
+
+We MAY use GitHub's
+[`CODEOWNERS`](https://help.github.com/en/articles/about-code-owners) feature
+to model the approval policy.
+
+## Stage 2: Candidate Recommendation
+
+The changes requested by the specification are considered plausible and
+desirable.
+
+The specification document itself is technically complete. It defines wire
+level formats for interoperability, error codes, algorithms, data structures,
+heuristics, behaviours, etc., in a way that it is sufficient to enable
+contributors to develop an interoperable implementation.
+
+There is at least ONE implementation conforming to the specification. That
+implementation serves as the Reference Implementation.
+
+The promotion from a `Working Draft` to a `Candidate Recommendation` is done
+via a *Pull Request* that is reviewed by the libp2p community in no more than
+8 working days.
+
+## Stage 3: Recommendation
+
+There are at least TWO implementations conforming to the specification, with
+demonstrated cross-interoperability. This is the supreme stage in the
+lifecycle of a specification.
+
+The promotion from a `Candidate Recommendation` to a `Recommendation` is done
+via a *Pull Request* that is reviewed by the libp2p community in no more than
+8 working days.
+
+## Stage 4: Deprecated
+
+The specification is no longer applicable and the community actively
+discourages new implementations.
+
+Transition to this stage is usually the result of a new version of the
+specification reaching the `Candidate Recommendation` stage.
+
+The transition from a `Candidate Recommendation` or a `Recommendation` stage
+to the `Deprecated` stage is performed via a *Pull Request* that is reviewed
+by the libp2p community in no more than 5 working days.
+
+## Stage 5: Abandoned Draft
+
+In order to motivate accountability, efficiency and order, a specification
+that stays in the `Working Draft` stage for over 4 months of its initial
```

Comment:
4 months? That's just arbitrary and too soon for the velocity we have with those things.
Make it 6 months minimum, and not an automatic transition.

Reply:
That's where the extensions come in. I really don't want specs to linger for 6 months with zero activity. I'd rather notice this earlier and establish checkpoints for extension. A spec is given a cumulative runway of 12 months; I think it's reasonable.